release.sh

```diff
@@ -25,28 +29,54 @@ changelog() {
 }
 
 bump() {
-  # Bump package version and generate changelog
-  VERSION="${NEXT/v/}"
 
-  echo "Bump version to ${VERSION}"
+  if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+      printfln "Exiting for pull request without versioning."
+
+      exit 0;
+  fi
+
+  last_tag=$(git tag --sort=-creatordate | head -n 1)
+
+  printfln "$last_tag"
+
+  if [ "$BRANCH" == "master" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
+      echo "Bumping the version: ${last_tag} -> ${NEXT}"
+      git tag "${NEXT}"
+
+      hub release create "$NEXT" -m "$NEXT" || true
+  fi
+}
 
-  # Update version in the following files
-  sed -i "s/\(\"version\":\s*\"\)[^\"]*\(\"\)/\1${VERSION}\2/g" package.json
+compare_and_release() {
+  ## Compare the package.json file from two recent commits to master branch and export value to NEXT variable
+  ## if differs else NEXT=false
 
-  # Generate change log
-  changelog
-  echo ""
+  previous_commit_hash=$(git rev-parse @~)
 
-  # Generate new build.
-  yarn && yarn test && yarn build
+  git fetch --all
+  git checkout ${previous_commit_hash}
 
-  # Prepare to commit
-  git add CHANGELOG.md package.json yarn.lock && \
-    git commit -v --edit -m "${VERSION} Release :tada: :fireworks: :bell:" && \
-    git tag "$NEXT" && \
-    echo -e "\nRelease tagged $NEXT"
-  git push origin HEAD --tags
-  yarn publish --new-version "${VERSION}" --no-git-tag-version
+  old_version=$(cat package.json | jq -r ".version")
+
+  printfln "Old package version: ${old_version}"
+
+  git checkout master
+
+  new_version=$(cat package.json | jq -r ".version")
+
+  printfln "New package version: ${new_version}"
+
+  NEXT=false
+
+  if [ "$old_version" != "$new_version" ]; then
+    printfln "Publishing changes to npm with version: ${new_version}."
```

Comment:
Remove thoplo at the end because the version will look confusing.

```suggestion
    printfln "Publishing changes to npm with version: ${new_version}"
```

Reply:
ok :smile: